google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableExporterUtils.java

```diff
@@ -0,0 +1,234 @@
+    for (AttributeKey<?> attributeKey : attributes.keySet()) {
+      if (PROMOTED_RESOURCE_LABELS.contains(attributeKey)) {
+        monitoredResourceBuilder.putLabels(
+            attributeKey.getKey(), String.valueOf(attributes.get(attributeKey)));
+      } else {
+        if (attributes.get(attributeKey) != null) {
+          metricLabels.put(attributeKey.getKey(), String.valueOf(attributes.get(attributeKey)));
+        }
+      }
+    }
+    metricLabels.put(CLIENT_UID.getKey(), taskId);
+
+    TimeSeries.Builder builder =
+        TimeSeries.newBuilder()
+            .setResource(monitoredResourceBuilder.build())
+            .setMetricKind(convertMetricKind(metricData))
+            .setMetric(
+                Metric.newBuilder()
+                    .setType(metricData.getName())
+                    .putAllLabels(metricLabels.build())
+                    .build())
+            .setValueType(convertValueType(metricData.getType()));
+
+    TimeInterval timeInterval =
+        TimeInterval.newBuilder()
+            .setStartTime(convertTimestamp(pointData.getStartEpochNanos()))
+            .setEndTime(convertTimestamp(pointData.getEpochNanos()))
+            .build();
+
+    builder.addPoints(createPoint(metricData.getType(), pointData, timeInterval));
+
+    return builder.build();
+  }
+
+  static String getProjectId(PointData pointData) {
+    return pointData.getAttributes().get(PROJECT_ID);
+  }
+
+  private static MetricKind convertMetricKind(MetricData metricData) {
+    switch (metricData.getType()) {
+      case HISTOGRAM:
+      case EXPONENTIAL_HISTOGRAM:
+        return convertHistogramDataType(metricData.getHistogramData());
+      case LONG_GAUGE:
+      case DOUBLE_GAUGE:
+        return GAUGE;
+      case LONG_SUM:
+        return convertSumDataType(metricData.getLongSumData());
+      case DOUBLE_SUM:
+        return convertSumDataType(metricData.getDoubleSumData());
+      default:
+        return UNRECOGNIZED;
+    }
+  }
+
+  private static MetricKind convertHistogramDataType(HistogramData histogramData) {
+    if (histogramData.getAggregationTemporality() == AggregationTemporality.CUMULATIVE) {
+      return CUMULATIVE;
+    }
+    return UNRECOGNIZED;
+  }
+
+  private static MetricKind convertSumDataType(SumData<?> sum) {
+    if (!sum.isMonotonic()) {
+      return GAUGE;
+    }
+    if (sum.getAggregationTemporality() == AggregationTemporality.CUMULATIVE) {
+      return CUMULATIVE;
+    }
+    return UNRECOGNIZED;
+  }
+
+  private static ValueType convertValueType(MetricDataType metricDataType) {
+    switch (metricDataType) {
+      case LONG_GAUGE:
+      case LONG_SUM:
+        return INT64;
+      case DOUBLE_GAUGE:
+      case DOUBLE_SUM:
+        return DOUBLE;
+      case HISTOGRAM:
+      case EXPONENTIAL_HISTOGRAM:
+        return DISTRIBUTION;
+      default:
+        return ValueType.UNRECOGNIZED;
+    }
+  }
+
+  private static Timestamp convertTimestamp(long epochNanos) {
```

Comment:
I think protobuf already has a utility that does is `Timestamps.fromNanos`

[fromNanos](https://protobuf.dev/reference/java/api-docs/com/google/protobuf/util/Timestamps.html#fromNanos-long-)